felt/src/lib.rs

```diff
@@ -1348,14 +1354,12 @@ mod test {
             prop_assert!(sqrt < p, "{}", sqrt);
         }
 
-        // There is a known bug in this implementation where the squared root
-        // we compute is that of the underlying integer rather than that of the
-        // field element.
-        // See: https://github.com/lambdaclass/cairo-rs/issues/1076
-        // WIP fix: https://github.com/lambdaclass/cairo-rs/pull/1150
         #[test]
-        #[ignore]
         fn sqrt_is_inv_square(x in any::<Felt252>()) {
+            // In the regular use case the number upon which sqrt is called will always be a felt (aka a numer lower than cairo_prime)
+            // In order to get a number for which we know its root it is not a good enough solution to square the felt (as we might get something bigger than cairo prime)
+            // We can instead use the square root (using the biguint implementation), to guarantee that its square will be inside the felt range
+            let x = Felt252::from(x.to_biguint().sqrt());
```

Comment:
I'm not sure I follow. The type is `Felt252` so the square will be in range, as it is corrected after multiplication. And being the square of a field element makes it a quad residue, right?

Reply:
Fixed